stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -577,7 +577,7 @@ function (*)(A::AbstractQ, b::StridedVector)
     end
     lmul!(Anew, bnew)
 end
-function (*)(A::AbstractQ, B::StridedMatrix)
+function (*)(A::AbstractQ, B::AbstractMatrix)
```

Comment:
This might be bad if `B` is a large sparse matrix or similar… I suspect that it was in order to exclude that case that `StridedMatrix` was used here?

Reply:
I've dropped this and rebased the commits. This PR is now only concerned with improving qr! for generic matrices, but does not change the interface for * and [rl]mul!.